os/net/mac/tsch/tsch-queue.c

```diff
@@ -504,13 +504,19 @@ tsch_queue_backoff_inc(struct tsch_neighbor *n)
 {
   /* Increment exponent */
   n->backoff_exponent = MIN(n->backoff_exponent + 1, TSCH_MAC_MAX_BE);
+  /* Avoid potential 16-bit integer overflow */
+  if(n->backoff_exponent > 16) {
+    n->backoff_exponent = 16;
+  }
```

Comment:
Would it be clearer for users to move this compile-time with something like:

```
#if TSCH_MAC_MAX_BE > 16
#error TSCH_MAC_MAX_BE must be 16 or lower
#endif
```

With the run-time check, abnormal large `TSCH_MAC_CONF_MAX_BE` will be masked at run-time and it may be difficult for users to understand the results.

Reply:
Updated and rebased the PR, please take a look!

Reply:
:+1: